From ea30964a2be5485c1e4e693777526f17820b8a26 Mon Sep 17 00:00:00 2001
From: Jian Zhang <zhangjian210@huawei.com>
Date: Mon, 19 Sep 2022 02:46:59 +0000
Subject: [PATCH] [Huawei] bugfix for RAS acpi map
 
Offering:ascend
CVE:
Reference:
Type:feature
reason: when using dts support RAS, in linux5.10 we can't using acpi-map directly.
In this case, if the memory has been maped, we just return it's virtual memory
address.
 
Signed-off-by: Jian Zhang <zhangjian210@huawei.com>
---
 arch/arm64/kernel/acpi.c | 5 +++++
 1 file changed, 5 insertions(+)
 
diff --git a/arch/arm64/kernel/acpi.c b/arch/arm64/kernel/acpi.c
index a81105cfe..203e3c950 100644
--- a/arch/arm64/kernel/acpi.c
+++ b/arch/arm64/kernel/acpi.c
@@ -25,6 +25,7 @@
 #include <linux/smp.h>
 #include <linux/serial_core.h>
 #include <linux/pgtable.h>
+#include <acpi/dt_apei.h>
 
 #include <acpi/ghes.h>
 #include <acpi/processor.h>
@@ -268,6 +269,10 @@ void __iomem *acpi_os_ioremap(acpi_physical_address phys, acpi_size size)
 	efi_memory_desc_t *md, *region = NULL;
 	pgprot_t prot;
 
+	/* For normal memory we already have a cacheable mapping. */
+	if (memblock_is_map_memory(phys) && enable_acpi_dt_apei)
+		return (void __iomem *)__phys_to_virt(phys);
+
 	if (WARN_ON_ONCE(!efi_enabled(EFI_MEMMAP)))
 		return NULL;
 
-- 
2.17.1
 
